ctt/process_dependencies.py

```diff
@@ -577,96 +673,93 @@ def process_images(
     for target_cfg in processing_cfg['targets'].values():
         tgt_oci_registries.update(target_cfg['kwargs']['registries'])
 
+    replication_plan = ctt.model.ReplicationPlan()
+
     for tgt_oci_registry in tgt_oci_registries:
-        yield from process_images_for_target(
+        replication_plan_step = create_replication_plan_step(
             processing_cfg=processing_cfg,
-            component_descriptor_v2=component_descriptor_v2,
+            root_component_descriptor=component_descriptor_v2,
             component_descriptor_lookup=component_descriptor_lookup,
             tgt_oci_registry=tgt_oci_registry,
             tgt_ocm_repository=tgt_ocm_repository,
             oci_client=oci_client,
             delivery_service_client=delivery_service_client,
+            replication_mode=replication_mode,
+            component_filter=component_filter,
+            reftype_filter=reftype_filter,
+            remove_label=remove_label,
+        )
+        replication_plan.steps.append(replication_plan_step)
+
+    logger.info(replication_plan)
+
+    for replication_plan_step in replication_plan.steps:
+        yield from process_replication_plan_step(
+            replication_plan_step=replication_plan_step,
+            component_descriptor_v2=component_descriptor_v2,
```

Comment:
can we drop the `_v2`-suffix, while we are refactoring (if you remember, this v2-suffix dates back to very olden times, where not even cnudie was coined as a name)? :-)